cpp/src/arrow/util/rle_encoding.h

```diff
@@ -592,11 +592,12 @@ bool RleDecoder::NextCounts() {
       return false;
     }
     repeat_count_ = count;
-    // XXX (ARROW-4018) this is not big-endian compatible
+    T value = 0;
```

Comment:
This change comes from https://issues.apache.org/jira/browse/ARROW-4018?focusedCommentId=16932199&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-16932199